File: swimming-contest-software/src/main/java/domain/SwimmingDistances.java
package domain;

/**
 * Available swimming distances for races.
 */
public enum SwimmingDistances {
    _50,
    _200,
    _800,
    _1500;

    public static SwimmingDistances distanceFromInteger(Integer value) {
        return switch (value) {
            case 1 -> _50;
            case 2 -> _200;
            case 3 -> _800;
            case 4 -> _1500;
            default -> null;
        };
    }

    public static Integer integerFromDistance(SwimmingDistances swimmingDistance) {
        return switch (swimmingDistance) {
            case _50 -> 1;
            case _200 -> 2;
            case _800 -> 3;
            case _1500 -> 4;
        };
    }
}

File: MPP-lab-project/src/Domain/SwimmingDistances.cs
using System;

namespace MPP_lab_project.Domain;

public enum SwimmingDistances
{
    _50m,
    _200m,
    _800m,
    _1500m
}

public static class SwimmingDistancesMethods
{
    public static SwimmingDistances DistanceFromInteger(Int32 value) =>
        value switch 
        {
            1 => SwimmingDistances._50m,
            2 => SwimmingDistances._200m,
            3 => SwimmingDistances._800m,
            4 => SwimmingDistances._1500m,
            _ => throw new ArgumentOutOfRangeException(nameof(value), value, null)
        };

    public static Int32 IntegerFromDistance(SwimmingDistances swimmingDistances) =>
        swimmingDistances switch
        {
            SwimmingDistances._50m => 1,
            SwimmingDistances._200m => 2,
            SwimmingDistances._800m => 3,
            SwimmingDistances._1500m => 4,
            _ => throw new ArgumentOutOfRangeException(nameof(swimmingDistances), swimmingDistances, null)
        };
}
